devito/symbolics/printer.py

```diff
@@ -151,6 +152,19 @@ def _print_TrigonometricFunction(self, expr):
             func_name += 'f'
         return func_name + '(' + self._print(*expr.args) + ')'
 
+    def _print_Or(self, expr):
+        # Enforce parentheses if Or(And() ...)
+        # This is only to avoid compiler warnings, because technically
+        # && has precedence over || in C (see for example
+        # https://en.cppreference.com/w/c/language/operator_precedence)
+        args = []
+        for i in expr.args:
+            v = self._print(i)
+            if isinstance(i, And):
```

Comment:
Would probably generic check if Relational

Reply:
uhm, I'll think about this one